arbor/morph/locset.cpp

```diff
@@ -68,6 +68,32 @@ std::ostream& operator<<(std::ostream& o, const location_& x) {
     return o << "(location " << x.loc.branch << " " << x.loc.pos << ")";
 }
 
+// Wrap mlocation_list (not part of public API).
+
+struct location_list_: locset_tag {
+    explicit location_list_(mlocation_list ll): ll(std::move(ll)) {}
+    mlocation_list ll;
+};
+
+locset location_list(mlocation_list ll) {
+    return locset{location_list_{std::move(ll)}};
+}
+
+mlocation_list thingify_(const location_list_& x, const mprovider& p) {
+    auto n_branch = p.morphology().num_branches();
+    for (mlocation loc: x.ll) {
+        if (loc.branch>=n_branch) {
+            throw no_such_branch(loc.branch);
+        }
+    }
+    return x.ll;
+}
+
+std::ostream& operator<<(std::ostream& o, const location_list_& x) {
+    o << "(list";
```

Comment:
Should this print `(sum (location x y) (location a b) ....)` to would make the round trip to string and back again valid?

Reply:
Seems reasonable! Will edit.